### ISSUE TITLE
Feature: Multiple Ref-/VarPort connections

### ISSUE THREAD

**Issue by PhilippPlank** [lava-nc/lava#55]
Currently a RefPort can only be connected to exactly one VarPort (1:1 connection).

Enabling 1:many connections to set multiple Vars

Open questions:
- What should the encoding of the RefPort read look like, when receiving data from multiple Vars? (maybe [Var1, data1, Var2, data2, ...])
- How useful is this feature?

**Comment by awintel** [comment]
Reading from multiple Vars through one RefPort is likely not useful at all. But writing would just mean we duplicate the written data.

In turn this means, that connecting to multiple Vars implies that we cannot write on a RefPort.

**Comment by PhilippPlank** [comment]
Issue #59 should be done in parallel when implementing this issue.

**Comment by PhilippPlank** [comment]
> Reading from multiple Vars through one RefPort is likely not useful at all. But writing would just mean we duplicate the written data.
> 
> In turn this means, that connecting to multiple Vars implies that we cannot write on a RefPort.

You mean that a RefPort connected to multiple Vars should only be able to write data to the Vars, but not be able to read anything.

I assume the RefPort should then be able to connect to multiple Vars of different destination processes?

**Comment by awintel** [comment]
Yes, that seems conceivable once we have the corresponding infrastructure in place that reads would not be possible in this case.
